core/controllers/admin.py

```diff
@@ -782,88 +770,122 @@ def get(self):
     def put(self):
         username = self.payload.get('username')
         role = self.payload.get('role')
-        topic_id = self.payload.get('topic_id')
         user_settings = user_services.get_user_settings_from_username(username)
 
         if user_settings is None:
             raise self.InvalidInputException(
                 'User with given username does not exist.')
 
         if role == feconf.ROLE_ID_TOPIC_MANAGER:
-            if feconf.ROLE_ID_TOPIC_MANAGER not in user_settings.roles:
-                user_services.add_user_role(user_settings.user_id, role)
+            raise self.InvalidInputException(
+                'Unsupported role for the handler.')
```

Comment:
the --> this

Also, maybe append "please assign a specific topic ID instead" or similar. (Alternatively, leaving a comment in the code to explain is also fine.)

Reply:
changed the to this and added a code-comment